### DISCUSSION TITLE
Send the body with reqwest

### DISCUSSION THREAD

**Discussion by JakubWasilczyk-Alias2k** [tokio-rs/axum#2746] [Q&A]
### Summary

Hi, I'm trying to send the incoming request body (which in my case is a multipart) with reqwest, but it must be streamed as the body could be very big.
This is my current code:
```
pub async fn post(
    State(state): State<AppState>,
    request: Request,
) -> ApiResponse {
    let url = "some valid url";

    let body = request.into_body().into_data_stream();
    let req = state.client.post(url);
    let req_body = Body::wrap_stream(body);
    // let request = request.body(body);

    Ok(StatusCode::NOT_IMPLEMENTED.into_response())
}
```
But I get this error on the `Body::wrap_stream` line:
```
 │   E  `(dyn axum::body::HttpBody<Data = bytes::Bytes, Error = axum::Error> + std::marker::Send + 'static)` cannot be shared between threads safely rustc (E0277) [218, 38]
 │      the trait `std::marker::Sync` is not implemented for `(dyn axum::body::HttpBody<Data = bytes::Bytes, Error = axum::Error> + std::marker::Send + 'static)`, which is required by `axum::body::BodyDataStream: std::marker::Sync` 
 │      required for `std::ptr::Unique<(dyn axum::body::HttpBody<Data = bytes::Bytes, Error = axum::Error> + std::marker::Send + 'static)>` to implement `std::marker::Sync`
```

I tried several things already but I can't figure out the way to do it and why I get this error. Tried doing something similar to the [stream-to-file](https://github.com/tokio-rs/axum/blob/axum-v0.7.5/examples/stream-to-file/src/main.rs) example but I get the errors that body is not a map. And the [reqwest-response](https://github.com/tokio-rs/axum/blob/axum-v0.7.5/examples/reqwest-response/src/main.rs) is doing it the other way.

Can someone please point me out what am I doing wrong and/or point out a way to do it?

### axum version

0.7.5 -F multipart

### reqwest version

0.12.4 -F json,stream

**Comment by jplatte** [comment]
Check out the second example from https://github.com/tokio-rs/axum/issues/2767#issuecomment-2150559926, I think it solves the `Sync` issue.
